phangsPipeline/scTheoreticalNoise.py

```diff
@@ -0,0 +1,201 @@
+        beam_fwhm_arcsec = 33.0*(300./freq_ghz)
+        
+    output = make_theoretical_noise(
+        field_ra = field_ra, field_dec = field_dec,
+        beam_fwhm_arcsec = beam_fwhm_arcsec,
+        **kwargs)
+
+    return(output)
+
+def make_theoretical_noise(
+        field_ra = None, field_dec = None,
+        beam_fwhm_arcsec = None,
+        exposure = None, noiseamp = None,
+        template_hdr = None, pix_scale = None,
+        support_factor = 1.0,
+        outfile = None, overwrite = False,
+        asnoise = True, normalize = False,
+):
+    """Make a theoretical noise map from a set of field centers.
+
+    field_ra : RA of field centers
+
+    field_dec : Dec of field centers
+
+    beam_fwhm_arcsec : the FWHM of the fields in arcsec
+
+    exposure : the exposure time of the pointing. Assume fixed across
+    all pointings if not supplied.
+
+    noiseamp : the amplitude of the noise (e.g., due to Tsys
+    variations) at each pointing. Assumed fixed if across all
+    pointings not supplied.
+
+    template_hdr : a header used to construct the output map. If not
+    supplied, one is constructed from the field centers.
+
+    pix_scale : pixel scale of the output image. Only used if a new
+    header is constructed. Defaults to a fraction of the primary beam
+    if not supplied.
+
+    outfile : if supplied, the output map is written to disk in this
+    file.
+
+    overwrite : if True will overwrite previous images. Default False.
+
+    asnoise : returns the results as a predicted noise map. Otherwise,
+    returns an exposure map ( propto 1/noise^0.5).
+
+    """
+
+    # Make sure we have a working header
+    if template_hdr is None:
+
+        max_ra = np.nanmax(field_ra)
+        min_ra = np.nanmin(field_ra)
+        max_dec = np.nanmax(field_dec)
+        min_dec = np.nanmin(field_dec)
+
+        if pix_scale is None:
+            pix_scale = beam_fwhm_arcsec / 10. / 3600.
+
+        pad_arcsec = beam_fwhm_arcsec*2.0
+        
+        template_hdr = make_simple_header_from_box(
+            tlc_coord = (max_ra, max_dec), brc_coord = (min_ra, min_dec),
+            pad_arcsec = pad_arcsec, pix_scale_deg = pix_scale,
+        )
+        
+    # Get images of RA and Dec
+    ra_img, dec_img = make_axes(header=template_hdr)
+
+    # Make sure we have an initialized set of weights
+    if exposure is None:
+        exposure = np.ones_like(field_ra, dtype=float)
+
+    if noiseamp is None:
+        noiseamp = np.ones_like(field_ra, dtype=float)
+
+    # This is an effective integration map
+    weight = exposure/noiseamp**2
+    
+    # Loop over fields
+    exposure_map = np.zeros_like(ra_img, dtype=float)
+
+    # Beam size in sigma units
+    beam_sigma_arcsec = beam_fwhm_arcsec / 2.355
+
+    for this_ra, this_dec, this_weight in \
+        zip(field_ra, field_dec, weight):
+        
+        # Assume Euclidean
+        offset_arcsec = np.sqrt(np.cos(this_dec/180.*np.pi)**2*(ra_img - this_ra)**2 + \
```

Comment:
since you have astropy.units support imported, consider switching to `* u.deg).to(u.arcsec)`.

But this is minor/pedantic and you can ignore.